.vsts-ci.yml

```diff
@@ -44,6 +44,9 @@ variables:
   value: -sign /p:SignCoreSdk=true
 - name: _officialBuildProperties
   value: /p:DotNetPublishUsingPipelines=true /p:OfficialBuilder=Microsoft /p:OfficialBuildId=$(Build.BuildNumber)
+- ${{ if and(notin(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
+  - name: enableSourceIndex
```

Comment:
nit: Use an `_` in the variable name to indicate it is local to this file. This is just a convention, not a requirement.